public/pages/utils/anomalyResultUtils.ts

```diff
@@ -389,6 +387,8 @@ export const getAnomalySummaryQuery = (
     },
   };
 
+  // If querying RT results: remove any results that include a task_id, as this indicates
```

Comment:
Please also give some context of this commit.

Reply:
So before this, many of the query-building fns were using `getResultFilters()` to generate the entity filters (either by model id for multi-category, or by entity.0.value for single-category). The problem was that `getResultFilters` was used in functions shared by single-entity detectors, so HC-related filters were inadvertently being added to the queries for these detectors, which was returning empty results.

This commit changes the code such that the queries will only add these filters if the detector is at least a HC detector (has an entity list length > 0). It also renames the helper fn to make it more clear that this is about entity-related filters.